public/app/features/explore/LogsNavigation.tsx

```diff
@@ -0,0 +1,243 @@
+  const [expectedQueries, setExpectedQueries] = useState<DataQuery[]>([]);
+
+  useEffect(() => {
+    const newPage = { logsRange: visibleRange || absoluteRange, queryRange: absoluteRange };
+    // We want to start new pagination if queries change or if absolute range is different than expected
+    if (!isEqual(expectedRange, absoluteRange) || !isEqual(expectedQueries, queries)) {
+      setPages([newPage]);
+      setExpectedQueries(queries);
+    } else {
+      setPages((pages) => {
+        const pagesWithNoDuplicates = pages.filter((page) => !isEqual(newPage.queryRange, page.queryRange));
+
+        const newPagesArray = [...pagesWithNoDuplicates, newPage].sort((a, b) => {
+          if (logsSortOrder === LogsSortOrder.Ascending) {
+            return a.queryRange.to > b.queryRange.to ? 1 : -1;
+          }
+          return a.queryRange.to > b.queryRange.to ? -1 : 1;
+        });
+        return newPagesArray;
+      });
+    }
+    // We don't want to add expectedRange as we want to run this only is absolute and visible range changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [visibleRange, absoluteRange, logsSortOrder]);
+
+  useEffect(() => {
+    const index = pages.findIndex((page) => page.queryRange.to === absoluteRange.to);
+    setCurrentPageIndex(index);
+  }, [pages, absoluteRange]);
+
+  useEffect(() => {
+    const initialRange = absoluteRange.to - absoluteRange.from;
+    setRequestRange(initialRange);
+    // We want to set requestRange only when the first query is run
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const changeTime = ({ from, to }: AbsoluteTimeRange) => {
+    setExpectedRange({ from, to });
+    onChangeTime({ from, to });
+  };
+
+  const formatTime = (time: number) => {
+    return `${dateTimeFormat(time, {
+      format: systemDateFormats.interval.second,
+      timeZone: timeZone,
+    })}`;
+  };
+
+  const oldestLogsFirst = logsSortOrder === LogsSortOrder.Ascending;
+  const theme = useTheme();
+  const styles = getStyles(theme, oldestLogsFirst, loading);
+  return (
+    <div className={styles.navContainer}>
+      {/*
+       * We are going to have 2 buttons - on the top and bottom - Oldest and Newest.
+       * Therefore I have at the moment duplicated the same code, but in the future iteration, it ill be updated
+       */}
+      {oldestLogsFirst && (
+        <Button
+          data-testid="fetchLogsTop"
+          className={styles.navButton}
+          variant="secondary"
+          onClick={() => {
+            // the range is based on initally selected range
+            changeTime({ from: visibleRange.from - requestRange, to: visibleRange.from });
+          }}
+          disabled={loading}
+        >
+          <div className={styles.navButtonContent}>
+            {loading ? <Spinner /> : <Icon name="angle-up" size="lg" />}
+            <div>Older</div>
+            <div>logs</div>
+          </div>
```

Comment:
if you set
```
      white-space: normal;
```

in `navButtonContent` you might be able to not wrap each child in a div

Reply:
👍 🚀